src/DotNetty.Transport/Channels/ChannelOutboundBuffer.cs

```diff
@@ -607,10 +644,10 @@ internal void Close(ClosedChannelException cause)
                     {
                         ReferenceCountUtil.SafeRelease(e.Message);
                         Util.SafeSetFailure(e.Promise, cause, Logger);
-                        if (e.Promise != TaskCompletionSource.Void && !e.Promise.TrySetException(cause))
-                        {
-                            Logger.Warn($"Failed to mark a promise as failure because it's done already: {e.Promise}", cause);
-                        }
+                        //if (e.Promise != TaskCompletionSource.Void && !e.Promise.TrySetException(cause))
```

Comment:
remove instead

Reply:
it's ok ， in linux  socket send would not send all buffer ， so  after send bufferlist it need get left not sent buffer